sympy/solvers/ode.py

```diff
@@ -6577,7 +6577,7 @@ def _linear_2eq_order1_type2(x, y, t, r):
     """
     r['k1'] = -r['k1']; r['k2'] = -r['k2']
     if (r['a']*r['d'] - r['b']*r['c']) != 0:
-        x0, y0 = symbols('x0, y0', cls=Dummy)
+        x0, y0 = symbols('x0, y0')
```

Comment:
I think you want to keep the dummy symbols here, don't you? Otherwise if the expression contains `x0`, this will stop working.